scripts/deploy.js

```diff
@@ -0,0 +1,40 @@
+require('dotenv').config();
+const { exec } = require('child_process');
+
+const args = process.argv.slice(2).join(' ');
+
+// Determine whether the environment is dev, stage, or prod from the command line. According to the environment, 
+// set the bucket and distribution id accordingly.
+switch (args) {
+  case 'dev':
+    bucket = process.env.DEV_BUCKET;
+    distributionId = process.env.DEV_DISTRIBUTION;
+    break;
+  case 'stage':
+    bucket = process.env.STAGE_BUCKET;
+    distributionId = process.env.STAGE_DISTRIBUTION;
+    break;
+  case 'prod':
+    bucket = process.env.PROD_BUCKET;
+    distributionId = process.env.PROD_DISTRIBUTION;
+    break;
+  default:
+    console.error('Invalid environment');
+    return;
+}
+
+// make sure the bucket is updated before invalidating the cloudfront distribution
+exec(`aws s3 sync ./build/ s3://${bucket} --delete`, (err, stdout, stderr) => {
+  if (err) {
+    console.error(err);
+    return;
+  }
+  console.log(stdout);
+  exec(`aws cloudfront create-invalidation --distribution-id ${distributionId} --paths "/*"`, (err, stdout, stderr) => {
```

Comment:
@usydusgduy123 this is the command to create a cache invalidation for a cloudfront distribution. basically what happens is the files from my front end build are stored in an s3 bucket in aws. in order to protect those files, public access is restricted. instead, the files are distributed to edge servers that are geographically close to the user. this boosts performance and has built in ddos protection. 

in the step above on line 27, that is the command to push the build files to s3. the cloudfront command is nested inside it to make sure it doesn't run until the files are updated in s3.